src/Microsoft.DotNet.XHarness.iOS.Shared/TestExecutingResult.cs

```diff
@@ -13,29 +13,30 @@ public enum TestExecutingResult
     InProgress = 0x1,
     Finished = 0x2,
     Waiting = 0x4,
-    StateMask = NotStarted + InProgress + Waiting + Finished,
+    StateMask = NotStarted | InProgress | Waiting | Finished,
 
     // In progress state
-    Building = 0x10 + InProgress,
-    BuildQueued = 0x10 + InProgress + Waiting,
-    Built = 0x20 + InProgress,
-    Running = 0x40 + InProgress,
-    RunQueued = 0x40 + InProgress + Waiting,
-    InProgressMask = 0x10 + 0x20 + 0x40,
+    Building = 0x10 | InProgress,
```

Comment:
Why don't we have the basic flags (0x10, 0x20, 0x40) also denoted here? This makes the semantics of the compound flags really hard to understand. In other words, every compound flag (defined via the or'ing) should be expressed in terms of other named flags, not a "magic" constants.

Reply:
I am not sure I follow what you propose here?
The way I understand it as that `0x10` means `Building` which is also `InProgress`.

Reply:
I got that, although technically, Building is 0x11 which is composed of <something> and InProgress. What's this <something>? What I'm after here is avoiding using the non-defined constants for clarity. Also the "Built = 0x20 | InProgress" seems weird as I'd expect Built be also Finished, which is not the case here. To be concrete, I could think of something like:
```
Build = 0x10,
BuildInProgress = Build | InProgress,
BuildFinished = Build | Finished,
BuildWaiting = Build | Waiting,
BuildQueued = Build | Waiting | InProgress
...
```
Similarly, I'd see e.g., "BuildSuceeded" defined not as magic `0x10000` and Succeeded, but as `Build | Succeeded`.
In other words, we seem to have several dimensions (States, Stages {Build, Run, perhaps Launch} and Results) projected to one, but the dimensions seem to not be aligned / defined right.

Reply:
Gotcha. I agree that would be better, however this is coming from Xamarin and I don't think I can change this

Reply:
I can maybe refactor it a little though

Reply:
I tried to do what you propose and I think your example makes sense, however it is not applicable to the enum we have in the code.

You could sensibly apply your strategy to a couple maybe three of the values there but even then it would lead to an explosion of values of the enum which is probably counter-productive?

I also cannot reasonably apply it without refactoring the usage, example of some of the problems:
- You have `Succeeded = 0x100 | Finished` so you could factor out `Success = 0x100` and make `Succeeded = Success | Finished`
- However, `Success` would ideally belong in between these:
    ```
    NotStarted = 0,
    InProgress = 0x1,
    Finished = 0x2,
    Waiting = 0x4,
    ```
- But for that I would already have to change the value.
- Even if I did that, the original enum clashes later where this wouldn't make sense because for example `BuildSucceeded = 0x10000 | Succeeded` would ideally be something like `Build | Succeeded` but since we already have a new value for that `0x10000`, we cannot do that atm.

I think the whole enum would need to be built from ground up again to make proper sense but even then, I don't think the way it is used cares about something like "x is a Build" so we would only introduce an explosion of values in the enum and it might become quite hard to use.

Reply:
Thanks for such detailed answer. Since the enum is already existing and being used, I agree with not refactoring it. 

I think you understood well what I'm after here, so just whenever we are building complex flags, please be mindful about the dimensions we are trying so fit (such as type of operation, state, result, etc.) as it's easy to clutter this as above.